upstairs/src/lib.rs

```diff
@@ -684,6 +688,14 @@ where
                         return Ok(())
                     }
                     Some(Message::Imok) => {}
+                    Some(Message::ReadOnlyMismatch(downstairs_read_only)) => {
+                        // Upstairs will never be able to connect, bail
+                        bail!(
+                            "downstairs read_only is {}, ours is {}!",
+                            downstairs_read_only,
+                            read_only,
+                        );
+                    }
                     Some(Message::YesItsMe(version)) => {
```

Comment:
I wonder if `read_only` should be decided here?  If an upstairs wants read write, and the downstairs is read only (or the other way around) we could abort at this point and not even need to wait for an activation request to show up as it will fail.

I also wonder if (in either this case or error after PromoteToActive) if the upstairs/downstairs will just get in a retry loop forever.

Reply:
I agree with this, and think it should extend to encrypted / non-encrypted status too.